commit ccb7b328361ae6d1cf4e907429381ae37c81520d
Author: Hua Su <suhua.tanke@gmail.com>
Date:   Tue Dec 11 20:16:18 2018 +0100

    lightnvm: pblk: add lock protection to list operations
    
    [ Upstream commit fde201a466c6ad5efd72cb54fdf2cefa8b6c6ad7 ]
    
    Protect the list_add on the pblk_line_init_bb() error
    path in case this code is used for some other purpose
    in the future.
    
    Signed-off-by: Hua Su <suhua.tanke@gmail.com>
    Reviewed-by: Javier González <javier@cnexlabs.com>
    Signed-off-by: Matias Bjørling <mb@lightnvm.io>
    Signed-off-by: Jens Axboe <axboe@kernel.dk>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/lightnvm/pblk-core.c b/drivers/lightnvm/pblk-core.c
index 2940cdc87af1..95be6e36c7dd 100644
--- a/drivers/lightnvm/pblk-core.c
+++ b/drivers/lightnvm/pblk-core.c
@@ -1252,15 +1252,22 @@ int pblk_line_recov_alloc(struct pblk *pblk, struct pblk_line *line)
 
 	ret = pblk_line_alloc_bitmaps(pblk, line);
 	if (ret)
-		return ret;
+		goto fail;
 
 	if (!pblk_line_init_bb(pblk, line, 0)) {
-		list_add(&line->list, &l_mg->free_list);
-		return -EINTR;
+		ret = -EINTR;
+		goto fail;
 	}
 
 	pblk_rl_free_lines_dec(&pblk->rl, line, true);
 	return 0;
+
+fail:
+	spin_lock(&l_mg->free_lock);
+	list_add(&line->list, &l_mg->free_list);
+	spin_unlock(&l_mg->free_lock);
+
+	return ret;
 }
 
 void pblk_line_recov_close(struct pblk *pblk, struct pblk_line *line)
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1127_ccb7b328361ae6d1cf4e907429381ae37c81520d.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1127_ccb7b328361ae6d1cf4e907429381ae37c81520d.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
